### PR TITLE
create '.vscode' directory before creating 'tasks.json'

### PR DESCRIPTION
**Pull request by elahehrashedi**
bugfix: https://github.com/microsoft/vscode-cpptools/issues/4280
this bug fix is merged with this code as well (there was a conflict): Adds a JSON parsing library for tasks.json https://github.com/microsoft/vscode-cpptools/pull/5954

In both of these below cases, the .vscode folder will be created if it doesn't exist (before creating tasks.json file):
a) running "Run and Debug"
or
b) running "C/C++ build and debug active file"

tested on VsCode v 1.48.2

### CODE REVIEW

File: Extension/src/common.ts [hunk 1]
```diff
@@ -633,6 +640,18 @@ export function readFileText(filePath: string, encoding: string = "utf8"): Promi
 
 /** Writes content to a text file */
 export function writeFileText(filePath: string, content: string, encoding: string = "utf8"): Promise<void> {
+    const folders = filePath.split(path.sep).slice(0, -1);
```
Comment (elahehrashedi) [inline Extension/src/common.ts]: creating folders in the path

File: Extension/src/common.ts [hunk 2]
```diff
@@ -61,59 +62,65 @@ export function getRawPackageJson(): any {
...
 
     rawTasksJson.version = "2.0.0";
 
-    const selectedTask2: vscode.Task = selectedTask;
-    if (!rawTasksJson.tasks.find((task: any) => task.label === selectedTask2.definition.label)) {
-        const task: any = {
-            ...selectedTask2.definition,
-            problemMatcher: selectedTask2.problemMatchers,
-            group: { kind: "build", "isDefault": true }
+    // Modify the current default task
```
Comment (elahehrashedi) [inline Extension/src/common.ts:107]: this needed to be changed

File: Extension/src/common.ts [hunk 3]
```diff
@@ -62,58 +62,66 @@ export function getRawPackageJson(): any {
     return rawPackageJson;
 }
 
-export function getRawTasksJson(): Promise<any> {
-    return new Promise<any>((resolve, reject) => {
-        const path: string | undefined = getTasksJsonPath();
+export async function getRawTasksJson(): Promise<any> {
+    const path: string | undefined = getTasksJsonPath();
+    return new Promise<any>(async (resolve, reject) => {
```
Comment (bobbrow) [inline Extension/src/common.ts]: When a function is marked `async`, you don't explicitly return a `Promise<>`.  You just `return {...}` or `throw new Error(...)`

**Review by bobbrow — CHANGES_REQUESTED** [review]

File: Extension/src/common.ts (hunk 3 above)
Comment (elahehrashedi) [reply Extension/src/common.ts]: I revert it to the previous version. I made this change while merging with another conflict.

**Review by sean-mcmanus — APPROVED** [review]

**Review by bobbrow — APPROVED** [review]

**Pull request merged**